ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -521,13 +521,22 @@ private StreamToTableJoiner(
 
       final JoinKey joinKey = joinNode.joinKey;
 
+      final FormatInfo leftValueFormatInfo = joinNode
+          .getLeft()
+          .getLeftmostSourceNode()
+          .getDataSource()
+          .getKsqlTopic()
+          .getValueFormat()
+          .getFormatInfo();
+
       switch (joinNode.joinType) {
         case LEFT:
           if (joinKey.isForeignKey()) {
             return leftTable.foreignKeyLeftJoin(
                 rightTable,
                 ((ForeignJoinKey) joinKey).getForeignKeyColumn(),
-                contextStacker
+                contextStacker,
+                leftValueFormatInfo
```

Comment:
nit: use `JoiningNode.getValueFormatForSource(joinNode.left).getFormatInfo()` rather than duplicating the code above.